docs/subnets/introduction-to-vm.md

```diff
@@ -0,0 +1,333 @@
+
+- Define the representation of a blockchain's state
+- Represent the operations in that state
+- Apply the operations in that state
+
+Each block in the blockchain contains a set of state transitions. Each block is applied in order from the blockchain's initial genesis block to its last accepted block to reach the latest state of the blockchain.
+
+## Blockchain
+
+A blockchain has 2 components: **Consensus Engine** and **Virtual Machine**. VMs mainly deal with the implementation related to the block's structure: building and parsing. The Consensus Engine helps in reaching consensus on the block built by the VM. Here is a brief overview:
+
+1. A node wants to update the blockchain's state
+2. The node's VM will notify the consensus engine that it wants to update the state
+3. Consensus engine will request the block from the VM with updates
+4. Consensus engine will verify the returned block and rely on VM's implementation of `verify()` method
+5. Consensus engine will gossip the successfully verified blocks within the network to reach consensus on this block
+6. Depending upon the consensus results, the engine can either accept or reject the block
+7. Every virtuous node on the network should have the same verdict for a particular block
+8. It depends on the VM implementation on what to do with the accepted or rejected block
+
+AvalancheGo provides the consensus engine for every blockchain on the Avalanche Network. The consensus engine relies on the VM interface to handle building, parsing, and storing blocks as well as verifying and executing on behalf of the consensus engine.
+
+## How to Load a VM
+
+VMs are created as a module, whose binary is registered by a node running `AvalancheGo`, against the **vmID** (binary file name must be vmID). VMID is a string that is zero-extended to a 32-byte array and encoded in CB58.
+
+In order to load a VM, its binary must be put it the proper place where `AvalancheGo` can find and load it. See [here](../nodes/maintain/avalanchego-config-flags.md#--build-dir-string) for more details. There could be multiple VM plugins in this directory.
+
+A VM can be initialized as blockchains. A blockchain can run as a separate process from AvalancheGo and can communicate with `AvalancheGo` over gRPC. This is enabled by `rpcchainvm`, a special VM that uses [`go-plugin`](https://pkg.go.dev/github.com/hashicorp/go-plugin) and wraps another VM implementation. The C-Chain, for example, runs the [Coreth](https://github.com/ava-labs/coreth) VM in this fashion.
+
+### API Handlers
+
+We interact with a blockchain and underlying VM through API handlers. Specifically, these handlers are implemented as **Services**.
+
+Handlers serve the response for the incoming HTTP requests; and handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as `AvalancheGo`. VM implements 2 kinds of handlers:
```

Comment:
`with **gRPC** for efficiently making calls from other services` or `with **gRPC** for efficiently responding to calls from other services` ?

Reply:
How about this - 

with **gRPC** for efficiently communicating with other services...

Reply:
OK. For my understanding, are Handlers making call to or being called from other services?

Reply:
AFAIK, handlers are being called from other services (here avalanchego)